packages/asset-swapper/src/utils/market_operation_utils/orders.ts

```diff
@@ -203,19 +206,39 @@ function createBridgeOrder(fill: CollapsedFill, opts: CreateOrderFromPathOpts):
     const bridgeAddress = getBridgeAddressFromSource(fill.source, opts);
 
     let makerAssetData;
-    if (isCurveSource(fill.source)) {
-        const { curveAddress, fromTokenIdx, toTokenIdx, version } = getCurveInfo(fill.source, takerToken, makerToken);
-        makerAssetData = assetDataUtils.encodeERC20BridgeAssetData(
-            makerToken,
-            bridgeAddress,
-            createCurveBridgeData(curveAddress, fromTokenIdx, toTokenIdx, version),
-        );
-    } else {
-        makerAssetData = assetDataUtils.encodeERC20BridgeAssetData(
-            makerToken,
-            bridgeAddress,
-            createBridgeData(takerToken),
-        );
+    switch (fill.source) {
+        case ERC20BridgeSource.CurveUsdcDai:
+        case ERC20BridgeSource.CurveUsdcDaiUsdt:
+        case ERC20BridgeSource.CurveUsdcDaiUsdtTusd:
+        case ERC20BridgeSource.CurveUsdcDaiUsdtBusd:
+        case ERC20BridgeSource.CurveUsdcDaiUsdtSusd:
+            const { curveAddress, fromTokenIdx, toTokenIdx, version } = getCurveInfo(fill.source, takerToken, makerToken);
+            makerAssetData = assetDataUtils.encodeERC20BridgeAssetData(
+                makerToken,
+                bridgeAddress,
+                createCurveBridgeData(curveAddress, fromTokenIdx, toTokenIdx, version),
+            );
+            break;
+        case ERC20BridgeSource.UniswapV2:
+            makerAssetData = assetDataUtils.encodeERC20BridgeAssetData(
+                makerToken,
+                bridgeAddress,
+                createUniswapV2BridgeData([makerToken, takerToken]),
+            );
+            break;
+        case ERC20BridgeSource.UniswapV2Eth:
+            makerAssetData = assetDataUtils.encodeERC20BridgeAssetData(
+                makerToken,
+                bridgeAddress,
+                createUniswapV2BridgeData([makerToken, takerToken]),
```

Comment:
should this be `[makerToken, WETH, takerToken]`?

Reply:
aaaah good catch, thanks!